contributors/design-proposals/cluster-info-configmap.md

```diff
@@ -0,0 +1,25 @@
+# Persist cluster-wide data in ConfigMap
+
+## Abstract
+
+We want to allow cluster-wide data to be persisted for general purposes, currently for us (service catalog) is to focus on the ability to identify different clusters that access our service broker with a 'cluster-id'.
+
+## Motivation
+
+Currently there is no way for service catalog api server to identify the different clusters that access our service brokers. The goal of this design is to have a unify storage for cluster related information. The information should be retained and survive between component failures. 
+
+## Detailed design
+
+A configmap `cluster-info` is created in the `default` namespace at the starup of the controller. The initial content of the map will be as the following:
+
+```
+apiVersion: v1
+data:
+  cluster-id: uuid-of-the-cluster
```

Comment:
- if there's ever a difference between the Controller.configId and the configMaps then the configMap wins - it is the source of truth
- if someone changes the configMap then that new value becomes the source of truth and during the next reconcile loop the controller will be updated
- the uuid is random and only created by the reconcile loop when it detects the configMap (or clusterID in the CM) is missing
- we're assuming the uuid code generates unique ID
- not sure what you mean by "where is this id denormalized to?" - can you elaborate?
- "use of this id?" - can you elaborate? In most cases people can view it as just a unique string to identify one cluster from another.